src/java.base/share/classes/java/lang/Class.java

```diff
@@ -186,6 +187,30 @@
  * a class or interface is hidden has no bearing on the characteristics
  * exposed by the methods of class {@code Class}.
  *
+ * <h2><a id=unnamedClasses>Unnamed Classes</a></h2>
+ *
+ * A {@code class} file representing an unnamed class is generated by
+ * a Java compiler from a source file for an unnamed class. The {@code
+ * Class} object representing an unnamed class is top-level,
+ * {@linkplain #isSynthetic synthetic}, and {@code final}. While an
+ * unnamed class does <em>not</em> have a name in its Java source
+ * form, the various name-related methods of {@code java.lang.Class}
+ * do return non-null and non-empty results for the {@code Class}
+ * object representing an unnamed class.
+ *
+ * Conventionally, a Java compiler, starting from a source file for an
+ * unnamed class, say {@code HelloWorld.java}, creates a
+ * similarly-named {@code class} file, {@code HelloWorld.class}, where
+ * the class stored in that {@code class} file is named {@code
+ * "HelloWorld"}, matching the base names of the source and {@code
+ * class} files.
+ *
+ * For the {@code Class} object for an unnamed class {@code
+ * HelloWorld}, the methods to get the {@linkplain #getName name},
+ * {@linkplain #getTypeName type name}, {@linkplain #getSimpleName
+ * simple name}, and {@linkplain #getCanonicalName canonical name} all
+ * return results equal to "{@code HelloWorld}".
```

Comment:
I recommend moving the quotes into `{@code}` because this is how the file name is represented above. Same is used for existing strings in code, like the rendering of `{@value #STRING_FIELD}` values.
```suggestion
 * return results equal to {@code "HelloWorld"}.
```